### PR TITLE
Fixes issue #251 : App Crashing on touching near right and bottom border in camera fragment 

### PR DESCRIPTION
**Pull request by vinaysajjanapu**
Fixes issue #251 

Changes: Previously, for calculating focus area fixed values(2000x1000) were used. So I modified the function such that values = device's screen size and it solved the issue.

### CODE REVIEW

**Comment by phimpme-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:tada: Thanks for the PR @vinaysajjanapu we hope you have changed the variables in the upload_apk_forks.sh its for easy moderation.</td>
    </tr>
  </tbody>
</table>

:link: Maintainers: the apk to test can be found at https://github.com/vinaysajjanapu/phimpme-android/blob/apk/app-debug.apk

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by vinaysajjanapu** [comment]
@Abhi2424shek @pa1pal please review.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 1]
```diff
@@ -651,6 +654,8 @@ public void onPause(){
 	Camera mCamera;
 	private static  final int FOCUS_AREA_SIZE= 300;
 
+	int screen_width = 900,screen_height = 1600;
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Aren't you initializing the variable here? If yes then Pls don't give it a value.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 2]
```diff
@@ -708,8 +713,15 @@ private void focusOnTouch(MotionEvent event) {
 
         //Calculate the Focus Area
         private Rect calculateFocusArea(float x, float y) {
-            int left = clamp(Float.valueOf((x / mSurfaceView.getWidth()) * 2000 - 1000).intValue(), FOCUS_AREA_SIZE);
-            int top = clamp(Float.valueOf((y / mSurfaceView.getHeight()) * 2000 - 1000).intValue(), FOCUS_AREA_SIZE);
+			//getting screen size
+			Display display = ((WindowManager) getContext().getSystemService(Context.WINDOW_SERVICE)).getDefaultDisplay();
+			Point size = new Point();
+			display.getSize(size);
+			screen_width = size.x;
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Use the functions which are already in `Utils.java` [Link](https://github.com/fossasia/phimpme-android/blob/development/app/src/main/java/vn/mbm/phimp/me/utils/Utils.java#L154-L164)

Comment (vinaysajjanapu) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: fine

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java (hunk 1 above)
Comment (vinaysajjanapu) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: I used them for testing and forgot to remove them.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 3]
```diff
@@ -708,8 +713,12 @@ private void focusOnTouch(MotionEvent event) {
 
         //Calculate the Focus Area
         private Rect calculateFocusArea(float x, float y) {
-            int left = clamp(Float.valueOf((x / mSurfaceView.getWidth()) * 2000 - 1000).intValue(), FOCUS_AREA_SIZE);
-            int top = clamp(Float.valueOf((y / mSurfaceView.getHeight()) * 2000 - 1000).intValue(), FOCUS_AREA_SIZE);
+			//getting screen size
+			screen_width = Utils.getScreenWidth(getContext());
+			screen_height = Utils.getScreenHeight(getContext());
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: International error.

Comment (vinaysajjanapu) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: sorry.. I didn't get you. can you please elaborate? 
As the code which I wrote is functional, I don't understand what you are talking about.

Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Ah sorry my keyboard auto corrected. It was `intendation error`. Apologies.

Comment (vinaysajjanapu) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: OK fine.........if that matters I will change it. 
By the way, it is indentation, not intendation.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 4]
```diff
@@ -708,10 +713,14 @@ private void focusOnTouch(MotionEvent event) {
 
         //Calculate the Focus Area
         private Rect calculateFocusArea(float x, float y) {
-            int left = clamp(Float.valueOf((x / mSurfaceView.getWidth()) * 2000 - 1000).intValue(), FOCUS_AREA_SIZE);
-            int top = clamp(Float.valueOf((y / mSurfaceView.getHeight()) * 2000 - 1000).intValue(), FOCUS_AREA_SIZE);
+			//getting screen size
+		screen_width = Utils.getScreenWidth(getContext());
+		screen_height = Utils.getScreenHeight(getContext());
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Please use 4 spaces and not tabs or 8 spaces.

Comment (vinaysajjanapu) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: done

File: app/src/main/java/vn/mbm/phimp/me/PhimpMe.java [hunk 5]
```diff
@@ -63,7 +67,8 @@
 //        forceCloseDialogAfterToast = false,
 //        resToastText = R.string.crash_report_text)
 @SuppressWarnings("deprecation")
-public class PhimpMe extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener //, android.view.GestureDetector.OnGestureListener
+public class PhimpMe extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener,
+        ViewPager.OnPageChangeListener //, android.view.GestureDetector.OnGestureListener
```
Comment (pa1pal) [inline app/src/main/java/vn/mbm/phimp/me/PhimpMe.java]: @vinaysajjanapu is this the code for adding swipe gesture? These changes were already reverted. Rebase your code to current development branch.
Make sure it show only the relevant additions.

Comment (vinaysajjanapu) [inline app/src/main/java/vn/mbm/phimp/me/PhimpMe.java]: @pa1pal I rebased the code and it completely got messed up. 
Shall I close this and open a new PR

please neglect my latest commits to this.

**Comment by vinaysajjanapu** [comment]
Sorry. My Mistake. I messed up things here trying to rebase the branch.
So, I want to open a new PR for this.

**Pull request closed without merging**